tests/test_updater_with_simulator.py

```diff
@@ -19,6 +19,23 @@
 from tests import utils
 from tests.repository_simulator import RepositorySimulator
 
+# DataSet is only here so type hints can be used:
+# It is a dict of name to test dict
+DataSet = Dict[str, str]
+
+# Test runner decorator: Runs the test as a set of N SubTests,
+# (where N is number of items in dataset), feeding the actual test
+# function one test case at a time
+def run_sub_tests_with_dataset(dataset: DataSet):
```

Comment:
Instead of copying the decorator code twice, it would be better to move it to a dedicated tests/util/helper/etc file and reuse it.

Reply:
I will change my commit description to clarify that the descriptor is similar, not the same with the one in `test_metadata_serialization.py`.
First, the `DataSet` type here should be `DataSet = Dict[str, Tuple[str, str]]` instead of `DataSet = Dict[str, str]` which I would fix.
Second, the decorator annotations (including the nested two functions) have different annotations than the ones inside `test_metadata_serialization.py`.

One solution in order to reuse the decorator will be to remove the annotations, but I think annotations are crucial in understanding how the decorator works and I think they should exist and be correct.

Reply:
This is one of the few places where `Any` annotation makes sense: the dict values are specific to the test and the decorator does not care what the actual type is.

Reply:
What about the other annotations? I mean the ones we are currently using:
`function: Callable[["TestUpdater", str], None]` and `test_cls: "TestUpdater"`.
Do you think we should use `Any` here too?
As I mentioned above my only concern is if it will be understandable for a new contributor.

Reply:
There's nothing in those functions that actuall requires a `TestUpdater`, AFAICT. A `unittest.TestCase` should work just fine?

Reply:
I guess you are right, those classes inherit `unittest.TestCase`, so this can be generalized and still give information with the annotations.